lib/network_monitor.c

```diff
@@ -893,6 +893,13 @@ static void send_getaddr_command(void *user_data)
 {
         struct mptcpd_nm *const nm = user_data;
 
+        /*
+          Don't bother attempting to retrieve IP addresses if no
+          network interfaces are being tracked.
+         */
+        if (l_queue_isempty(nm->interfaces))
```

Comment:
It looks like this is depending on the `NULL` assignment in `mptcpd_nm_destroy()`, but if that was assigned then `nm` has already been freed prior to getting dereferenced here.

Reply:
I'm not sure I follow.  `nm` hasn't been freed yet at this point.  Only `nm->interfaces` has been freed.  This function gets called in two cases:

1. Immediately after network interface retrieval through `rtnetlink` completes.
2. During destruction of the `l_netlink` object used to retrieve the network interfaces and addresses.  In this case, the `l_netlink` object only get destroyed after thiis function returns.

In both cases, `nm` hasn't been freed yet.  `nm` is freed after the `l_netlink` object, i.e. `nm->rtnl` is destroyed.

Reply:
Ok, I was only thinking of it running asynchronously and didn't realize it would get invoked during `l_netlink_destroy()`

Reply:
Ideally this code shouldn't run during `l_netlink_destroy()` at all, but the workaround for sequential `l_netlink_send()` calls currently forces mptcpd to do so.  I'll try to come up with an alternative approach.